repo-scripts/repo-toolbox/src/ReadmeAction.ts

```diff
@@ -0,0 +1,140 @@
+      documentation: "Use this to update the repo's README.md"
+    });
+  }
+
+  private static _isPublished(project: RushConfigurationProject): boolean {
+    return project.shouldPublish || !!project.versionPolicyName;
+  }
+
+  protected async onExecute(): Promise<void> {
+    // abstract
+
+    const rushConfiguration: RushConfiguration = RushConfiguration.loadFromDefaultLocation();
+
+    const repoReadmePath: string = path.resolve(rushConfiguration.rushJsonFolder, 'README.md');
+    const existingReadme: string = await FileSystem.readFileAsync(repoReadmePath);
+    const generatedProjectSummaryStartIndex: number = existingReadme.indexOf(
+      GENERATED_PROJECT_SUMMARY_START_COMMENT_TEXT
+    );
+    const generatedProjectSummaryEndIndex: number = existingReadme.indexOf(
+      GENERATED_PROJECT_SUMMARY_END_COMMENT_TEXT
+    );
+
+    if (generatedProjectSummaryStartIndex === -1 || generatedProjectSummaryEndIndex === -1) {
+      throw new Error(
+        `Unable to find "${GENERATED_PROJECT_SUMMARY_START_COMMENT_TEXT}" or ` +
+          `"${GENERATED_PROJECT_SUMMARY_END_COMMENT_TEXT}" comment in "${repoReadmePath}"`
+      );
+    }
+
+    const readmePrefix: string = existingReadme.substr(
+      0,
+      generatedProjectSummaryStartIndex + GENERATED_PROJECT_SUMMARY_START_COMMENT_TEXT.length
+    );
+
+    const readmePostfix: string = existingReadme.substr(generatedProjectSummaryEndIndex);
+
+    const builder: StringBuilder = new StringBuilder();
+    const orderedProjects: RushConfigurationProject[] = [...rushConfiguration.projects];
+    Sort.sortBy(orderedProjects, (x) => x.projectRelativeFolder);
+
+    builder.append(readmePrefix);
+    builder.append('\n');
+    builder.append('\n');
+    builder.append('## Published Packages\n\n');
+    builder.append('<!-- the table below was generated using the ./repo-scripts/repo-toolbox script -->\n\n');
+    builder.append('| Folder | Version | Changelog | Package |\n');
+    builder.append('| ------ | ------- | --------- | ------- |\n');
+    for (const project of orderedProjects.filter((x) => ReadmeAction._isPublished(x))) {
+      // Example:
+      //
+      // | [/apps/api-extractor](./apps/api-extractor/)
+      // | [![npm version](https://badge.fury.io/js/%40microsoft%2Fapi-extractor.svg
+      //     )](https://badge.fury.io/js/%40microsoft%2Fapi-extractor)
+      // | [changelog](./apps/api-extractor/CHANGELOG.md)
+      // | [@microsoft/api-extractor](https://www.npmjs.com/package/@microsoft/api-extractor)
+      // |
+
+      const scopedName: string = project.packageName; // "@microsoft/api-extractor"
+      const folderPath: string = project.projectRelativeFolder; // "apps/api-extractor"
+      let escapedScopedName: string = scopedName; // "%40microsoft%2Fapi-extractor"
+      escapedScopedName = Text.replaceAll(escapedScopedName, '/', '%2F');
+      escapedScopedName = Text.replaceAll(escapedScopedName, '@', '%40');
```

Comment:
What is the advantage of using `Text.replaceAll` here instead of something like:
```typescript
const escapedScopedName: string = scopedName.replaceAll('/', '%2f').replaceAll('@', '%40');
```

Or how about using encodeURIComponent?
```typescript
const escapedScopedName: string = encodeURIComponent(scopedName);
```